### PR TITLE
Make PiecewiseBilinear::parse a static function

### LINKED ISSUE #23023: Make parse in PiecewiseBilinear static function
## Reason

Allows other objects to use the same parsing capability and csv file format.
I want to use the function in an application class.

## Design

`data_file_name` becomes an argument, `paramErrors` become `mooseErrors`

## Impact

Slightly less instructive error messages emitted from `PiecewiseBilinear::parse`

### PR DESCRIPTION
**Pull request by snschune**
closes #23023 

## Reason

To be able to use the same parse method in other classes. 

## Design

`data_file_name` becomes method parameter and `paramError` become `mooseError`

## Impact

Line numbers of input errors are not emitted in error handling.

### CODE REVIEW

File: framework/include/functions/PiecewiseBilinear.h [hunk 1]
```diff
@@ -63,6 +63,11 @@ class PiecewiseBilinear : public Function
   virtual Real value(Real t, const Point & pt) const override;
   virtual ADReal value(const ADReal & t, const ADPoint & pt) const override;
 
+  static void parse(const std::string & data_file_name,
+                    std::vector<Real> & x,
+                    std::vector<Real> & y,
+                    ColumnMajorMatrix & z);
```
Comment (GiudGiud) [inline framework/include/functions/PiecewiseBilinear.h:69]: you can add a std::string parameter object_name which will just be populated by a call to name() on most objects to keep the informative error messages

**Comment by GiudGiud** [comment]
@snschune do you still need this?

**Comment by GiudGiud** [comment]
I put up a patch for Falcon. Even if we decide we dont need this, the patch is a good idea. 
https://github.com/idaholab/falcon/pull/124

**Comment by GiudGiud** [comment]
Patch is merged.

**Comment by lindsayad** [comment]
I'm waiting to hear from @snschune before reviewing

**Comment by snschune** [comment]
Yes, I need this for a PR in Pronghorn.
https://github.inl.gov/ncrc/pronghorn/pull/932

**Review by lindsayad — APPROVED** [review]
I don't like removing `paramError` but I think with the object name and that data file name, there is still enough information for users to figure out errors.

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**